tests/integration/rdb.tcl

```diff
@@ -6,13 +6,16 @@ set server_path [tmpdir "server.rdb-encoding-test"]
 exec cp tests/assets/encodings.rdb $server_path
 exec cp tests/assets/list-quicklist.rdb $server_path
 
-start_server [list overrides [list "dir" $server_path "dbfilename" "list-quicklist.rdb"]] {
+foreach {type max_listpack_size} {listpack -1 quicklist 0} {
```

Comment:
When `list_max_listpack_size` is 0, list listpack encoding will be disabled (count in quicklistSizeAndCountLimit will be 0), I will add comments in redis.conf

Reply:
> please make sure that code is reachable in the test suite and comment / resolve this comment.

I've seen it fully covered in gcov before, but let's add a more precise test to cover it.

Reply:
ok, so `list_max_listpack_size` of `0` means a single element per quicklist node (same as `1`), but also implies it'll always be quicklist?
it seems odd to me (should be an invalid config, but since we didn't block that config in the past, we can't start now.
but i think it's effects should be the same as setting it to 1.

IIUC the difference from the current code is that it'll still use listpack if there's a single element in the list.
any downside in that?

Reply:
> please make sure that code is reachable in the test suite and comment / resolve this comment.

That code has been converted by `LREM starting from tail with negative count (2)` test.

Reply:
Please see commit [`6dfb7bf` (#11303)](https://github.com/redis/redis/pull/11303/commits/6dfb7bf9349e9b8282c096e902f41978a5ce9c0b)

Set `fill` to 1 when `fill` is 0, also optimize the original code that limits the quicklist node(_quicklistNodeAllowInsert and _quicklistNodeAllowMerge).

Reply:
i still don't like the odd case of 0 (should be an invalid config, and if it isn't, then it should be the same as it was before (synonym for 1).

maybe we should accept the fact that a list with a single element of one byte will always be a listpack, and it's ok that we have no control over it...
maybe for the vast majority of the tests that runs on the two variants it doesn't have an impact?

Reply:
Yes, Indeed, I'm trying to modify it in a new way.

Reply:
I unified the limit method(`quicklistNodeMeetLimit`) of quicklist node and listpack.
It means that now quicklist has add two new interfaces to the outside.

There are two main modifications to `list-listpack-max-size`
1. When 0, it works the same as 1
2. When < -5, it works the same as -5

Reply:
> When < -5, it works the same as -5, if we dont do that, < -5 means disable list listpack

i'm not sure i understand, is that different than what we had before?